### DISCUSSION TITLE
Room for external holder with thicker base middle

### DISCUSSION THREAD

**Discussion by ja2ui0** [joshreve/dactyl-keyboard#12] [Ideas]
I want to produce a bottom plate that nests slightly inside the outer shell and screws in. This is so that the base itself contributes to the overall stability (feeling) of the model, rather than relying on the screws alone to serve this function. Setting the parameter `base_thickness` to 7 (using all other defaults) results in a base middle that is 2mm thicker than the outer edge with the screw holes, so that's good.

However, if you select the external holder option to use loligagger's holders, the resulting hole is now 2mm too short (because of the increased base thickness) and the holder won't fit. For now, it's easy enough to sketch a rectangle in freecad and cut a 2mm deep hole in the rendered base right behind the hole to accommodate the holder. However, anyone trying to accomplish the same thing by this method will run into the same problem.

My suggestion is to include a rectangular section slightly deeper and wider than loligagger's holder model which will NOT increase in thickness (ie: be included as part of `base_rim_thickness`) so that in no cases will a thicker base middle impede the placement of the external holder. This change should have no practical cosmetic or functional impact on any other use case for the thicker base middle.

**Comment by joshreve** [comment]
Interesting.  That is something to look at.  I have no dimensions for the actual holder at the moment, but could theoretically add the holder depth and width clearance to a plate cutout.  I had thought of the situation of increasing the middle of the plate to go up into the case, but hadn't thought about the tray aspect.  I'll take a look when I get a chance.  May be an easy addition.  You may also run into thumb cluster interferences if you don't extent the height of the overall keyboard to get a straight wall in that area (depending on your cluster choice).

**Reply by ja2ui0** [reply]
Good callout, I hadn't thought about that. I just had a really rough measure, and I think the default cluster can handle a 2mm thickness for the middle of the plate if the pins are oriented at the upward slope. Any other way, and at least one switch pin would have to bend.
